### DISCUSSION TITLE
MonitorControl v4.2.0

### DISCUSSION THREAD

**Discussion by JoniVR** [MonitorControl/MonitorControl#1425] [Release Discussion]
### Improvements
* DDC support for high-end M2 (Pro/Max/Ultra) systems by @waydabber, maker of **[BetterDisplay](https://betterdisplay.pro/)** in https://github.com/MonitorControl/MonitorControl/pull/1404
* Fixed "Possible memory leak in Arm64ddc? #1087" by @waydabber, maker of **[BetterDisplay](https://betterdisplay.pro/)** in https://github.com/MonitorControl/MonitorControl/pull/1181

### Translations & other
* cs_CZ: update Main.strings for 4.1.0 by @isametry in https://github.com/MonitorControl/MonitorControl/pull/1085
* Fixed Readme typo by @prabch in https://github.com/MonitorControl/MonitorControl/pull/1156
* Fixed typo (cs_cz) by @isametry in https://github.com/MonitorControl/MonitorControl/pull/1249
* Fixed typo by @tentena in https://github.com/MonitorControl/MonitorControl/pull/1256
* Updated README for Homebrew by @Goooler in https://github.com/MonitorControl/MonitorControl/pull/1317
* Fixed typo in Italian translation by @AlexPerathoner in https://github.com/MonitorControl/MonitorControl/pull/1325
* Updated Korean translation by @zzulu in https://github.com/MonitorControl/MonitorControl/pull/1358
* String changed, since it changed in System Settings by @Herrner in https://github.com/MonitorControl/MonitorControl/pull/1353
* Fixed typo in README by @Gelbotron in https://github.com/MonitorControl/MonitorControl/pull/1375
* Updated Simplified Chinese and Traditional Chinese translations by @wjy2311077 in https://github.com/MonitorControl/MonitorControl/pull/1417
* Updated Simplified Chinese Translations by @Jerry23011 in https://github.com/MonitorControl/MonitorControl/pull/1146
* Refined Simplified Chinese localization by @Jerry23011 in https://github.com/MonitorControl/MonitorControl/pull/1319
* Russian translation - @sertrsv by @sertrsv in https://github.com/MonitorControl/MonitorControl/pull/1075

## New Contributors
* @sertrsv made their first contribution in https://github.com/MonitorControl/MonitorControl/pull/1075
* @isametry made their first contribution in https://github.com/MonitorControl/MonitorControl/pull/1085
* @Jerry23011 made their first contribution in https://github.com/MonitorControl/MonitorControl/pull/1146
* @prabch made their first contribution in https://github.com/MonitorControl/MonitorControl/pull/1156
* @tentena made their first contribution in https://github.com/MonitorControl/MonitorControl/pull/1256
* @Goooler made their first contribution in https://github.com/MonitorControl/MonitorControl/pull/1317
* @AlexPerathoner made their first contribution in https://github.com/MonitorControl/MonitorControl/pull/1325
* @Herrner made their first contribution in https://github.com/MonitorControl/MonitorControl/pull/1353
* @Gelbotron made their first contribution in https://github.com/MonitorControl/MonitorControl/pull/1375

**Full Changelog**: https://github.com/MonitorControl/MonitorControl/compare/v4.1.0...v4.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/MonitorControl/MonitorControl/releases/tag/v4.2.0'>MonitorControl v4.2.0</a>.</em>

**Comment by markwoitaszek** [comment]
thank you for this build. I just discovered this app 3 days ago, looking for something better than Dell's DDC app and was surprised Monitor Control did not work. I had seen the M2 yml file in the code but then noticed the latest release was a year ago. Came back today to clone & build but was pleasantly surprised it was done! Perfect timing!

**Comment by devabhixda** [comment]
Well this version doesn't work on my M2 pro running 13.5.2 (22G91) with my Samsung monitor. It makes the display go blank if I try adjusting brightness. Rolling back to 4.1.0

**Reply by the0neyouseek** [reply]
Your monitor may not be compatible with DDC control, try disabling it in the app settings

**Comment by wclyffe4** [comment]
Love this app but having an issue with the latest build (4.20). I was using the last version (4.10) with my MacBookPro M2 and an older LED Cinema display with no issues, but now my monitor will not come to full brightness. The light sensor on the LED Cinema display seems to be affected because when I shine a light at the sensor the display brightens up. Any thoughts about what changed this setting and how to readjust it?

**Reply by waydabber** [reply]
Previous versions were unable to control the cinema display properly by hardware means. The new version can, however you'll need to set the max DDC range to 255. Search discussions for it, there are many threads dealing with this! :)

**Reply by wclyffe4** [reply]
Thanks for your help, I'll give it a search and hope I understand it!!

**Reply by wclyffe4** [reply]
@waydabber I just added the max range to 255 in the advanced settings and it works beautifully! Thank you.

**Comment by Jestblanq** [comment]
Hello, it is not working to control the volume on LG 29WN ultrawide monitor, shows that the volume is changing but it's not really doing anything, am I missing a setting or something?

**Reply by jack20412** [reply]
hi I am facing the same problem with u , do u have any idea why this happen or do u have any solution.

**Comment by Yepitsmeposting** [comment]
Hello,

I'm on a 2023 M2 Max studio and for some reason the Monitor Control App will not automatically open after turning on the computer. This is despite Monitor Control being allowed to open in the settings and the "start at login" checkbox being selected in the app. Every time I restart and open the app the checkbox is unchecked

**Comment by DoctorCaffeine** [comment]
You are a Hero for making this app! Running well new-install on M1 pro. Wondering though, is there a way to decouple brightness on keyboard from monitor? In the past just used hdmi external and MacBook as secondary monitor. Keyboard brightness would control the Macbook and would leave the hdmi display alone.   Now I can only seem to control Both macbook/external (usb-c) or neither.

**Reply by isametry** [reply]
Try and play with the Keyboard settings in MonitorControl. The `Screen to control` preference might just do what you're looking for?
![Snímek obrazovky 2024-02-01 v 19 47 51](https://github.com/MonitorControl/MonitorControl/assets/46103746/d92d5ca8-d008-458d-8336-e7ee6fa770b5)

Alternatively, you could set custom keyboard shortcuts for MonitorControl, which will adjust your external display, and keep the normal shortcuts for your built-in display. 

The latter is what I do with my setup – I set MonitorControl to `Custom keyboard shortcuts` and chose `⌘F14` and `⌘F15` for those. This is only for my external monitor. I configured MonitorControl to leave my built-in screen alone, and there, I just use the normal system shortcuts (`F1` and `F2`).

**Comment by lschumann1978** [comment]
I just want to say thank you to all who have worked on this app!! This is exactly what I was looking for for my M2 Pro Mac Mini and LG 40WP95C-W monitor. I'm software developer myself but this kind of work is way out of my league.

**Comment by Kokomodo** [comment]
I used MonitorControl 4.1.0 to change the DDC max from 100 to 255 for a 27" LED cinema display.  I had to use my 2013 macbook pro with a miniDisplay port and then connect the monitor to my mac Studio.  I did this for two monitors and then adjusted the color balance between them on the Studio.  I recently noticed that one of the monitors was significantly dimmer than the other.  I decided to use Monitor Control again to ensure that nothing had been reset on the monitors.  I updated to 4.2.0 and opened the app.  The only functioning controls on the Displays tab now are "Enable keyboard control for display" and "Brightness, Volume, and Contrast" radio buttons on the lower left of advanced settings.  I cannot enable "Use hardware DDC control" and cannot change the DDC read polling mode.  I cannot enter a new DDC max value.  Is this degradation in functionality related to the 4.1 to 4.2 update?  I tried reverting to 4.1.0 but see the same behavior.  Any ideas what is happening?

**Reply by Kokomodo** [reply]
Well, I don't know what happened, but I plugged in the other display (identical) and Monitor Control came up with "use hardware DDC control" enabled and all other controls functional.  I set the DDC max to 255 and that fixed the issue.  I then plugged in the first monitor I had set, and it ALSO came up with Use hardware DDC control enabled and full functionality.  No idea what changed.  This is all with 4.1.0, by the way.

**Reply by Kokomodo** [reply]
Finally, I switched the monitors over to the mac Studio (the last time I did this the DDC max setting persisted.  This time they dimmed again.  However, with Monitor Control 4.2 I was able to set the DDC max over the Thunderbolt port (through a USB-C to miniDisplay port adapter, where as in MC 4.1.0 I was not able to make such adjustments through that port.  Bizarre.  I'm not technical enough to know what the difference is, but both monitors appear to be functioning perfectly now.

**Comment by wclyffe4** [comment]
The problem with brightness control resolved for me with version v4.20. I can control the brightness with my keyboard and I get full brightness on the monitor.On Mar 22, 2024, at 7:14 AM, jack20412 ***@***.***> wrote:﻿
hi I am facing the same problem with u , do u have any idea why this happen or do u have any solution.

**Comment by artgoldberg** [comment]
Very nice! Thanks. I'll use MonitorControl v4.2.0 half a dozen times a day! You can be proud of your work.
Arthur

**Comment by thesagarmahajan** [comment]
This changed my life. Thanks.

**Comment by Philirr** [comment]
Guys, thank you so much, it is such a gamechanger! I start working when it's dark outside and end working when it's dark. During the day my room has changing lighting conditions and before I had to go through many painful button presses and slow menu navigation on my Philipps display to change brightness. Now it's sooo easy ❤️

**Comment by vidursaini** [comment]
I downloaded the Monitor Control App for my LG Ultrawide monitor but it doesn't let me adjust the speaker volume for some reason. No option to enable disable volume control in the app so that is probably not an issue. Anyone else facing this problem?

P.S: Monitor brightness working like a charm.
